### PR TITLE
Avoid redundant metadata reads in `ZarrArrayWrapper`

### LINKED ISSUE #8290: Potential performance optimization for Zarr backend
### What is your issue?

We have identified an inefficiency in the way the `ZarrArrayWrapper` works.  This class currently stores a reference to a `ZarrStore` and a variable name

https://github.com/pydata/xarray/blob/75af56c33a29529269a73bdd00df2d3af17ee0f5/xarray/backends/zarr.py#L63-L68

When accessing the array, the parent group of the array is read and used to open a new Zarr array.

https://github.com/pydata/xarray/blob/75af56c33a29529269a73bdd00df2d3af17ee0f5/xarray/backends/zarr.py#L83-L84

This is a relatively metadata-intensive operation for Zarr. It requires reading both the group metadata and the array metadata. Because of how this wrapper works, these operations currently happen _every time data is read from the array_. If we have a dask array wrapping the zarr array with thousands of chunks, these metadata operations will happen within every single task. For high latency stores, this is really bad.

Instead, we should just reference the `zarr.Array` object directly within the `ZarrArrayWrapper`. It's lightweight and easily serializable. There is no need to re-open the array each time we want to read data from it. This change will lead to an immediate performance enhancement in all Zarr operations.

### PR DESCRIPTION
**Pull request by olimcc**
Modify ZarrArrayWrapper to avoid re-reading metadata each time data is read from the array.

- [x] Closes #8290
- [x] Tests added
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`
- [ ] New functions/methods are listed in `api.rst`

### CODE REVIEW

File: xarray/tests/test_backends.py [hunk 1]
```diff
@@ -2862,6 +2863,47 @@ def create_zarr_target(self):
...
+        This test targets logic in backends.zarr.ZarrArrayWrapper, asserting that calls
+        to retrieve chunk data after initialization do not trigger additional
+        metadata requests.
+
+        https://github.com/pydata/xarray/issues/8290
+        """
+
+        import zarr
+
+        ds = xr.Dataset(data_vars={"test": (("Z"), np.array([np.nan]).reshape(1))})
```
Comment (jhamman) [inline xarray/tests/test_backends.py:2880]:
```suggestion
        ds = xr.Dataset(data_vars={"test": (("Z", ), np.array([123]).reshape(1))})
```

**Review by jhamman — APPROVED** [review]
Small suggestion but this looks great. No doubt this is going to really improve performance when reading data with zarr.

**Review by headtr1ck — APPROVED** [review]
I think this looks good, thanks!

Maybe some `get_array()` calls can be replaced by `_array`, but that's really minor.

You should add an entry in whats-new!

And as a bonus you could add a benchmark, but that's really optional (seems like there are only netcdf Io benchmarks right now).

**Comment by olimcc** [comment]
Thanks @headtr1ck!

> You should add an entry in whats-new!

Great, this has been done.

> And as a bonus you could add a benchmark, but that's really optional (seems like there are only netcdf Io benchmarks right now).

I didn't do this. I'll try to add a note to #8290 of a before/after scenario, to illustrate the impact of the reduced requests.

**Pull request merged**